numpyro/contrib/tfp/distributions.py

```diff
@@ -105,18 +107,71 @@ def _transform_to_bijector_constraint(constraint):
     return BijectorTransform(constraint.bijector)
 
 
+def _onehot_enumerate_support(self, expand=True):
+    n = self.event_shape[-1]
+    values = jnp.identity(n, dtype=jnp.result_type(self.dtype))
+    values = values.reshape((n,) + (1,) * len(self.batch_shape) + (n,))
+    if expand:
+        values = jnp.broadcast_to(values, (n,) + self.batch_shape + (n,))
+    return values
+
+
 class _TFPDistributionMeta(type(NumPyroDistribution)):
+    @lru_cache(maxsize=None)
     def __getitem__(cls, tfd_class):
         assert issubclass(tfd_class, tfd.Distribution)
 
+        tfd_class_name = tfd_class.__name__
+
         def init(self, *args, **kwargs):
+            warnings.warn(
+                "Importing distributions from numpyro.contrib.tfp.distributions is "
+                "deprecated. You should import distributions directly from "
+                "tensorflow_probability.substrates.jax.distributions instead.",
+                FutureWarning,
+            )
             self.tfp_dist = tfd_class(*args, **kwargs)
 
         init.__signature__ = inspect.signature(tfd_class.__init__)
 
-        _PyroDist = type(tfd_class.__name__, (TFPDistribution,), {})
+        _PyroDist = type(tfd_class_name, (TFPDistribution,), {})
         _PyroDist.tfd_class = tfd_class
```

Comment:
I think `tfd_class` is not used elsewhere, could you help me remove this and the one in TFPDistribution? Thanks!